### ISSUE TITLE
$http get-request should append [] to name when arrays

### ISSUE THREAD

**Issue by filipbech** [angular/angular.js#11162]
```js
$http({
  url:'/api',
  params: {
    arr:[1,2,3]
  }
});
```

Currently gives you a get request to (wrong)
/api/?arr=1&arr=2&arr=3

should give you (expected)
/api/?arr%5B%5D=1&arr%5B%5D=2&arr%5B%5D=3
(when decoded its: /api/?arr[]=1&arr[]=2&arr[]=3)

Current workaround is to append [] to the attributename like:
```js
params: {
  'arr[]':[1,2,3]
}
```

but that seems weird and broken?

**Comment by pkozlowski-opensource** [comment]
This is duplicate of https://github.com/angular/angular.js/issues/3740 which is going to be solved in the 1.4 time-frame.

**Comment by filipbech** [comment]
perfect thankyou! you are totally right! 
I did a search for it but that issue didn't come up...
